examples/maps/maps.bal

```diff
@@ -1,43 +1,73 @@
+    // Declare a `map` constrained by the type `string`.
     map<string> m;
 
-    // You can also declare and initialize a `map` with a map literal.
+    string city = "Colombo";
+    string country = "Sri Lanka";
+    string codeLiteral = "code";
+
+    // You can also declare and initialize a `map` with a mapping constructor
+    // expression.
     map<string> addrMap = {
+        // A field can be specified as a key-value pair.
         line1: "No. 20",
         line2: "Palm Grove",
+        // The value in a key-value pair can be any expression that evaluates
+        // to a value of a type that belongs to the constraint type of the map.
         city: "Colombo 03",
-        country: "Sri Lanka"
+        // A field can also be just a variable reference, which would result
+        // in the variable name being the field name and the variable itself
+        // being the value expression.
+        // This is equivalent to saying `country: country`.
+        country,
+        // The key in a key-value pair can also be a computed key.
+        // A computed key is defined by specifying the key expression within
+        // brackets. The key expression must belong to type `string`.
+        // For a computed key, the key expression is evaluated at runtime and
+        // the resulting value is used as the key.
+        [codeLiteral]: "00300"
     };
     io:println(addrMap);
 
-    // Retrieves a value using member access.
-    string country = <string>addrMap["country"];
-    io:println(country);
+    // Retrieve a value using member access.
+    // Member access returns the value if a field exists in the map with the
+    // specified key, or `()` if a field does not exist with the specified key.
+    // The type of a member access expression for a map is thus the union of
+    // the constraint type and the nil type.
+    string? countryValue = addrMap["country"];
+
+    if (countryValue is string) {
+        io:println(countryValue);
+    } else {
+        io:println("key 'country' not found");
+    }
 
-    // Retrieves a value using the `.get()` method.
-    // Panics if the map does not have a member with the specified key.
-    string line2 = addrMap.get("line2");
-    io:println(line2);
+    // Retrieve a value using the `.get()` method.
+    // If the map has a field with the specified key, `.get()` returns the value,
+    // or panics if the map does not have a field with the specified key.
+    // Thus, the type for `.get()` on a map is the map's constraint type.
+    string line2Value = addrMap.get("line2");
+    io:println(line2Value);
 
-    // The `.hasKey()` function checks if a map contains a specified key.
+    // The `.hasKey()` function checks if a map contains the specified key.
     boolean hasPostalCode = addrMap.hasKey("postalCode");
     io:println(hasPostalCode);
 
-    // Adds or updates the value of a key.
+    // Member access lvalue expressions can be used to add fields to a map
```

Comment:
```suggestion
    // Member access value expressions can be used to add fields to a map
```

Reply:
This is a term used in the spec. Added it to distinguish left hand side member access expressions vs right hand side.

Reply:
If we are using the term, we'll need to tell what an lvalue is too right? Should we have a separate BBE for that?

Reply:
I'm not quite sure, seems a bit advanced for a BBE. Let me rephrase this anyway.